test/controllers/income/estimatedPay/update/IncomeUpdateCalculatorControllerSpec.scala

```diff
@@ -83,30 +84,339 @@ class IncomeUpdateCalculatorControllerSpec
+
+      def payPeriodPage() = controller.payPeriodPage()(BuildFakeGet())
+
+      def workingHoursPage() = controller.workingHoursPage()(BuildFakeGet())
+
+      def handleWorkingHours(request: FakeRequest[AnyContentAsFormUrlEncoded])  = controller.handleWorkingHours()(request)
+
+      def handlePayPeriod(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handlePayPeriod()(request)
+
+      def payslipAmountPage(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.payslipAmountPage()(request)
+
+      def handlePayslipAmount(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handlePayslipAmount()(request)
+
+      def taxablePayslipAmountPage(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.taxablePayslipAmountPage()(request)
+
+      def handleTaxablePayslipAmount(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handleTaxablePayslipAmount()(request)
+
+      def payslipDeductionsPage() = controller.payslipDeductionsPage()(BuildFakeGet())
+
+      def handlePayslipDeductions(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handlePayslipDeductions()(request)
+
+      def bonusPaymentsPage(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.bonusPaymentsPage()(request)
+
+      def handleBonusPayments(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handleBonusPayments()(request)
+
+      def bonusOvertimeAmountPage(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.bonusOvertimeAmountPage()(request)
+
+      def handleBonusOvertimeAmount(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handleBonusOvertimeAmount()(request)
+
+      def checkYourAnswersPage(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.checkYourAnswersPage()(request)
+
+      def estimatedPayPage(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.estimatedPayPage()(request)
+
+      def handleCalculationResult(request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handleCalculationResult()(request)
+
+      def editIncomeIrregularHours(incomeNumber: Int, request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.editIncomeIrregularHours(incomeNumber)(request)
+
+      def handleIncomeIrregularHours(employmentId: Int, request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.handleIncomeIrregularHours(employmentId)(request)
+
+      def confirmIncomeIrregularHours(employmentId: Int, request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.confirmIncomeIrregularHours(employmentId)(request)
+
+      def submitIncomeIrregularHours(employmentId: Int, request: FakeRequest[AnyContentAsFormUrlEncoded]) = controller.submitIncomeIrregularHours(employmentId)(request)
+    }
+
+    def setup() = new UnifiedHarness(new TestIncomeUpdateCalculatorController)
+
+    def setupOnPageLoad() = {
```

Comment:
This seems like a bit of a anti pattern here. With everything returning the same type with access to all the methods a developer could easily call `setupOnPageLoad()` and then make a call to `submitIncomeIrregularHours()`. Which is obviously wrong. 

Maybe a better abstraction would be to have a harness for each endpoint that could also handle generic setup as well

something like:

```
object OnPageLoadHarness {
  sealed class OnPageLoadHarness() {
    when(employmentService.employment(any(), any())(any()))
      .thenReturn(Future.successful(Some(defaultEmployment)))

     when(journeyCacheService.cache(any())(any())).thenReturn(Future.successful(Map.empty[String, String]))

    def onPageLoad(employerId: Integer = 1) = controller.onPageLoad(employerId)(BuildFakeGet())
  }

  def buildOnPageLoad(): OnPageLoadHarness = new OnPageLoadHarness()
}
```